### ISSUE TITLE
Automate the integration tests

### ISSUE THREAD

**Issue by tomasbjerre** [gmessner/gitlab4j-api#311]
I have noticed that the integration tests do not run without a prepared Gitlab server available.

Would be nice if this was automated with Maven.

I have been using this plugin to do this before: https://dmp.fabric8.io/
And I have some code for setting up GitLab here: https://github.com/tomasbjerre/jenkins-configuration-as-code-sandbox

**Comment by gmessner** [comment]
@tomasbjerre 
I've actually already done this, but have not had time to get it to a production ready state and checked in.  I just happen to be a big docker advocate, been working with it before it was even released.    You should take a look at some of the gists I have around docker (https://gist.github.com/gmessner).

I happen to be a bit busy right now but expect to have time to check this in in the next 30 days or so.

**Comment by gmessner** [comment]
@tomasbjerre 
I've checked in all the updates to support docker based integration testing and released version 4.10.9 with these changes.

I've done it in such a way that a pre-configured GitLab server can still be used, but the intent is to use an off the shelf gitlab-ce Docker image, which can be started independently of the integration tests.  To run the integration tests simply do this on the command line:
```
./mvnw verify -Dgitlab.skip-docker-start=false
```
This will automatically start and stop the docker container.  If you'd like to pre-start the docker container and leave it running, do the following:
```
./mvnw docker:start -Dgitlab.skip-docker-start=false
```
And then run the tests with either of the following commands:
```
./mvnw verify
./mvnw integration-test
```
Once down testing, you can shutdown the docker container as follows:
```
./mvnw docker:stop -Dgitlab.skip-docker-start=false
```

I've gone ahead and closed this issue. If you find any problems or have additional requests please feel free to re-open this issue.

**Comment by gmessner** [comment]
@tomasbjerre 
Just wanted to give you a heads up that a couple of subsequent releases have been made which have improved the docker based integration tests.  You should make sure to pull the latest code to check out the docker stuff.
